core/src/iosMain/kotlin/com/revenuecat/purchases/kmp/Purchases.ios.kt

```diff
@@ -418,3 +450,28 @@ public actual class Purchases private constructor(private val iosPurchases: IosP
     public actual fun setCreative(creative: String?): Unit =
         iosPurchases.setCreative(creative)
 }
+
+internal fun PurchasesConfiguration.storeKitVersionToUse(): StoreKitVersion {
```

Comment:
I'm a little bit torn that this is only `internal` to be able to test it haha. What if we make it `private` to `PurchasesConfiguration` and just set `PurchasesConfiguration.storeKitVersion` to whatever this function calculates, in the constructor? That way we can test it by reading `PurchasesConfiguration.storeKitVersion`. 

E.g. in `PurchasesConfiguration.kt`, something like:

```diff
 public class PurchasesConfiguration private constructor(
     public val apiKey: String,
     public val appUserId: String?,
     public val purchasesAreCompletedBy: PurchasesAreCompletedBy,
     public val userDefaultsSuiteName: String?,
-    public val storeKitVersion: StoreKitVersion,
+    storeKitVersion: StoreKitVersion,
     public val showInAppMessagesAutomatically: Boolean,
     public val store: Store?,
     public val diagnosticsEnabled: Boolean,
     public val dangerousSettings: DangerousSettings,
     public val verificationMode: EntitlementVerificationMode,
     public val pendingTransactionsForPrepaidPlansEnabled: Boolean?
 ) {
+    public val storeKitVersion: StoreKitVersion = storeKitVersionToUse(
+        purchasesAreCompletedBy, 
+        storeKitVersion
+    )
}
```